homeassistant/components/wallbox/coordinator.py

```diff
@@ -174,6 +207,73 @@ async def async_set_charging_current(self, charging_current: float) -> None:
         )
         await self.async_request_refresh()
 
+    @_require_authentication
+    def _get_sessions(
+        self,
+        charger_id: str,
+        start_date: datetime.datetime,
+        end_date: datetime.datetime,
+    ) -> list[WallboxEvent]:
+        """Get charging sessions between timestamps for Wallbox."""
+
+        def create_description(event: dict[str, Any]) -> str:
+            tzone = dt_util.get_default_time_zone()
+            description = f"Session ID: {event["id"]}\n"
+            description += f"Charger name: {event["attributes"]["charger_name"]}\n"
+            description += f"Serial number: {event["attributes"]["charger"]}\n"
+            description += f"Username: {event["attributes"]["user_name"]}\n"
+            description += f"Start: {datetime.datetime.fromtimestamp(
+                    event["attributes"]["start"], tzone
+                )}\n"
+            description += f"End: {datetime.datetime.fromtimestamp(
+                    event["attributes"]["end"], tzone
+                )}\n"
+            description += f"Time: {datetime.datetime.fromtimestamp(
+                    event["attributes"]["time"]) - datetime.datetime.fromtimestamp(0)}\n"
+            description += f"Energy: {event["attributes"]["energy"]}KWh\n"
+            description += f"MID energy: {event["attributes"]["mid_energy"]}KWh\n"
+            description += f"Session cost: {event["attributes"]["cost"]}{event["attributes"]["currency"]["code"]}\n"
+            return description
+
+        data = self._wallbox.getSessionList(charger_id, start_date, end_date)
+        tzone = dt_util.get_default_time_zone()
+        events: list[WallboxEvent] = [
+            WallboxEvent(
+                charger_name=event["attributes"]["charger_name"],
+                username=event["attributes"]["user_name"],
+                session_id=event["id"],
+                currency=event["attributes"]["currency"]["code"],
+                session_type="",
+                serial_number=event["attributes"]["charger"],
+                time=event["attributes"]["time"],
+                energy=event["attributes"]["energy"],
+                mid_energy=event["attributes"]["mid_energy"],
+                cost_kw=0,
+                session_cost=event["attributes"]["cost"],
+                start=datetime.datetime.fromtimestamp(
+                    event["attributes"]["start"], tzone
+                ),
+                end=datetime.datetime.fromtimestamp(event["attributes"]["end"], tzone),
+                summary=f"Charger {event["attributes"]["charger_name"]}: {event["attributes"]["energy"]}KWh",
+                description=create_description(event),
```

Comment:
see above, I would use an f string here, and skip the seperate function. Also, put some description stuff in location and don't put the start and end time in description (it's already in other properties).